src/RemoteInboxNotifications/DataSetupForProducts.php

```diff
@@ -0,0 +1,93 @@
+<?php
+/**
+ * Handles $data setup for products.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\PluginsProvider\PluginsProvider;
+use \Automattic\WooCommerce\Admin\RemoteInboxNotifications\SpecRunner;
+
+/**
+ * Handles $data setup for products.
+ */
+class DataSetupForProducts {
+	/**
+	 * Initialize the class
+	 */
+	public static function init() {
+		add_action( 'product_page_product_importer', array( __CLASS__, 'run_on_product_importer' ) );
+		add_action( 'transition_post_status', array( __CLASS__, 'run_on_transition_post_status' ), 10, 3 );
+	}
+
+	/**
+	 * Set initial $data values.
+	 *
+	 * @param $data object The $data object.
+	 */
+	public static function init_data( &$data ) {
+		$data->there_were_no_products = ! self::are_there_products();
+		$data->there_are_now_products = ! $data->there_were_no_products;
+	}
+
+	/**
+	 * Are there products query.
+	 *
+	 * @return bool
+	 */
+	private static function are_there_products() {
+		$product_query = new \WC_Product_Query( array( 'limit' => -1 ) );
+		$product_count = count( $product_query->get_products() );
```

Comment:
Same `WC_Product_Query` comments as here: https://github.com/woocommerce/woocommerce-admin/pull/4334#discussion_r431280201

Reply:
I think I've got it - de85c214